src/util/starter.ts

```diff
@@ -1,199 +1,100 @@
-                if (!error && response.statusCode === 200 && targetDir !== undefined) {
-                    vscode.window.showInformationMessage("Successfully generated MicroProfile starter project!");
-
-                    //unzip the file
-                    var extract = require('extract-zip');
-                    extract(targetDir + '/' + artifactId + '.zip', { dir: targetDir }, async function (err: any) {
-                        // extraction is complete. make sure to handle the err
-                        if (err !== undefined) {
-                            vscode.window.showErrorMessage("Could not extract the MicroProfile starter project.");
-                        } else {
-                            // open the unzipped folder in a new VS Code window
-                            let uri = vscode.Uri.file(targetDir + '/' + artifactId);
-                            var newWindow: boolean = false;
-                            if (vscode.workspace.workspaceFolders !== undefined) {
-                                newWindow = true;
-                            }
-                            await vscode.commands.executeCommand('vscode.openFolder', uri, newWindow);
-                        }
-
-                    });
-
-                } else {
-                    vscode.window.showErrorMessage("Could not generate an MicroProfile starter project.");
-                }
-            }
-
-            await request(options, callback).pipe(fs.createWriteStream(targetDir + '/' + artifactId + '.zip'));
-        } else {
-            vscode.window.showErrorMessage(error);
-        }
+    // ask user to select one of the servers that are avalaible for the version of mp they selected
+    const mpServer = await util.askForMPserver(mpConfigurations[mpVersion].supportedServers);
+    if (mpServer === undefined) {
+      return;
     }
-    request(supportMatrixOptions, getSupportMatrix);
 
-}
\ No newline at end of file
+    // ask user to pick a list of mp specifications to use for the given version of mp they selected
+    const allSupportedSpecs = mpConfigurations[mpVersion].specs;
+    const specDescriptions = mpSupportMatrix.descriptions;
+    const mpSpecifications = await util.askForMPSpecifications(allSupportedSpecs, specDescriptions);
+    if (mpSpecifications === undefined) {
+      return;
+    }
+
+    const targetFolder = await util.askForFolder({
+      openLabel: "Generate into this folder",
+    });
+    if (targetFolder === undefined) {
+      return;
+    }
+
+    const targetDirString = targetFolder.fsPath;
+
+    const requestPayload = {
+      groupId: groupId,
+      artifactId: artifactId,
+      mpVersion: mpVersion,
+      supportedServer: mpServer,
+      javaSEVersion: javaSEVersion,
+      selectedSpecs: mpSpecifications,
+    };
+
+    // location to download the zip file
+    const zipPath = targetDirString + "/" + artifactId + ".zip";
```

Comment:
perhaps instead of "/" we should use `path.sep` from [path module](https://www.google.com/search?q=require+path+node+js&oq=require+path+node+js&aqs=chrome..69i57j0l7.4071j0j7&sourceid=chrome&ie=UTF-8)

Reply:
Changed to use `path.join` which joins them using the platform-specific separator.